scheduler/scheduler.py

```diff
@@ -898,45 +892,68 @@ def _schedule_jobs_on_workers(self):
                 self._worker_type_to_worker_id_mapping[worker_type])
             worker_state[worker_type] = {
                 'worker_ids': worker_ids,
-                'reserved_worker_ids': set(),
+                'assigned_worker_ids': set(),
                 'server_id_ptr': 0,
-                'num_workers_assigned': 0,
             }
 
-        # Keep jobs on the same server if possible.
-        already_scheduled_jobs = set()
-        new_worker_types = {}
-        for worker_type in scheduled_jobs:
-            for (job_id, scale_factor) in scheduled_jobs[worker_type]:
-                new_worker_types[job_id] = worker_type
+        prev_worker_types = {}
         for (job_id, worker_ids) in self._current_worker_assignments.items():
-            current_worker_type = \
```

Comment:
Did you remove this code completely? Hard to tell because github's diff control is terrible

Reply:
Well you can think of 915 - 931 as the replacement for this